### PR TITLE
rename 'runtime' interface name to 'store'

### LINKED ISSUE #13: Consider renaming the 'runtime' interface to 'store' for consistency with other WASI proposals
Since originally suggesting `runtime` as the interface name, some naming conventions have emerged elsewhere in WASI, in particular, having the interface names be nouns instead of adjectives.  Also, wasi-keyvalue has `wasi:keyvalue/store` and there's a nice symmetry between that and `wasi:config/store`.

The original idea of `runtime` was to distinguish the runtime `get` lookup from a more declarative interface that allowed a component to statically enumerate the specific configuration values it needed as [value imports](https://github.com/WebAssembly/component-model/blob/main/design/mvp/Explainer.md#-value-definitions).  However, I think a "store" communicates this just as well and there are plenty of names left for declarative value imports.

### PR DESCRIPTION
**Pull request by Mossaka**
It renames:

`runtime` -> `store`
`config-error` -> `error`

and

delete the hand-written `config.md` in preference of the generated `imports.md` doc. Closes #13 

question: I remember we briefly discussed this but I forgot about the rationals behind the subtle difference between the proposal name "wasi-runtime-config" and the actual WIT package name "wasi:config", what are the reasons? Could we rename this repo to "wasi-config" but keeps runtime configuration as an option in the README? @thomastaylor312 @lukewagner

### CODE REVIEW

**Comment by lukewagner** [comment]
Thanks!  To your question: originally, there was a sketch to use value imports to import configuration values, with 1 value import per individual configuration value, allowing the names of required configuration values to be known declaratively, allowing various DX/security/perf improvements.  But that was a much bigger lift (for the tooling and runtime), so we instead prioritized adding a more-dynamic (runtime) configuration interface (which we'd need in any case, since sometimes you just don't know AOT what configuration you need), hence the name of the repo.  Whatever the history, renaming the repo to wasi-config to match `wasi:config` makes sense to me.  Eventually, I'd like to return to the declarative value-import approach, we could just add that to this same repo (alongside `wasi:config/store`).

File: imports.md [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 </ul>
 </li>
 </ul>
-<h2><a name="wasi:config_runtime_0.2.0_draft"></a>Import interface wasi:config/runtime@0.2.0-draft</h2>
+<h2><a id="wasi_config_store_0_2_0_draft"></a>Import interface wasi:config/store@0.2.0-draft</h2>
 <hr />
 <h3>Types</h3>
-<h4><a name="config_error"></a><code>variant config-error</code></h4>
+<h4><a id="error"></a><code>variant error</code></h4>
 <p>An error type that encapsulates the different errors that can occur fetching config</p>
```
Comment (devigned) [inline imports.md:13]:
```suggestion
<p>An error type that encapsulates the different errors that can occur fetching configuration values.</p>
```

File: imports.md [hunk 2]
```diff
@@ -31,19 +31,19 @@ For simplicity right now in supporting multiple implementations, it is being lef
 </ul>
 <hr />
 <h3>Functions</h3>
-<h4><a name="get"></a><code>get: func</code></h4>
+<h4><a id="get"></a><code>get: func</code></h4>
 <p>Gets a single opaque config value set at the given key if it exists</p>
```
Comment (devigned) [inline imports.md:35]:
```suggestion
<p>Gets a string configuration value if the key exists. If the key does not exist, the value returned will be None.</p>
```

File: imports.md [hunk 3]
```diff
@@ -31,19 +31,19 @@ For simplicity right now in supporting multiple implementations, it is being lef
...
+<li><a id="get.key"></a><code>key</code>: <code>string</code></li>
 </ul>
 <h5>Return values</h5>
 <ul>
-<li><a name="get.0"></a> result&lt;option&lt;<code>string</code>&gt;, <a href="#config_error"><a href="#config_error"><code>config-error</code></a></a>&gt;</li>
+<li><a id="get.0"></a> result&lt;option&lt;<code>string</code>&gt;, <a href="#error"><a href="#error"><code>error</code></a></a>&gt;</li>
 </ul>
-<h4><a name="get_all"></a><code>get-all: func</code></h4>
+<h4><a id="get_all"></a><code>get-all: func</code></h4>
 <p>Gets a list of all set config data</p>
```
Comment (devigned) [inline imports.md:45]:
```suggestion
<p>Gets a list of each configuration key and value.</p>
```

File: wit/store.wit [hunk 4]
```diff
@@ -18,8 +18,8 @@ interface runtime {
     get: func(
         /// A string key to fetch
         key: string
-    ) -> result<option<string>, config-error>;
+    ) -> result<option<string>, error>;
 
     /// Gets a list of all set config data
```
Comment (devigned) [inline wit/store.wit:23]:
```suggestion
    /// Gets a list of each configuration key and value.
```

**Review by devigned — CHANGES_REQUESTED** [review]
Just words, no code

**Review by devigned — APPROVED** [review]
Thank you for improving upon my suggestions. LGTM!

**Comment by Mossaka** [comment]
Could you please take a look? @thomastaylor312

**Review by thomastaylor312 — APPROVED** [review]

**Comment by thomastaylor312** [comment]
And I am all good with a repo rename

**Pull request merged**